cross/x264/Makefile

```diff
@@ -3,56 +3,52 @@ PKG_VERS = 20240512-stable
 PKG_EXT = tar.gz
 PKG_GIT_HASH = 4613ac3c15fd75cebc4b9f65b7fb95e70a3acce1
 PKG_DIST_NAME = $(PKG_NAME)-master.$(PKG_EXT)
-PKG_DIST_FILE = $(PKG_NAME)-$(PKG_GIT_HASH).$(PKG_EXT)
 PKG_DIST_SITE = https://code.videolan.org/videolan/x264/-/archive/$(PKG_GIT_HASH)
 PKG_DIST_FILE = $(PKG_NAME)-git$(PKG_GIT_HASH).$(PKG_EXT)
 PKG_DIR = $(PKG_NAME)-$(PKG_GIT_HASH)
 
 DEPENDS =
-OPTIONAL_DEPENDS = native/nasm
+
+# configure fails in "endian test"
+UNSUPPORTED_ARCHS = $(OLD_PPC_ARCHS)
 
 HOMEPAGE = https://www.videolan.org/developers/x264.html
 COMMENT  = x264 is a free software library and application for encoding video streams into the H.264/MPEG-4 AVC forma
 LICENSE  = GPLv2
 
-GNU_CONFIGURE = 1
-
-CONFIGURE_ARGS  = --enable-shared
-CONFIGURE_ARGS += --enable-pic
+# configure with "GNU_CONFIGURE = 1" complains: "Unknown option --build=i686-pc-linux, ignored"
+CONFIGURE_ARGS  = --prefix=$(INSTALL_PREFIX)
+CONFIGURE_ARGS += --enable-shared
 CONFIGURE_ARGS += --disable-opencl
-CONFIGURE_ARGS += --prefix=$(INSTALL_PREFIX)
-CONFIGURE_ARGS += --cross-prefix=$(TC_PATH)$(TC_PREFIX)
 
 include ../../mk/spksrc.cross-cc.mk
 
-ifeq ($(findstring $(ARCH),$(ARMv5_ARCHS) $(ARMv7_ARCHS) $(ARMv7L_ARCHS)),$(ARCH))
-ENV += x264_ARCH=ARM
-CONFIGURE_ARGS += --disable-asm
-endif
+CONFIGURE_ARGS += --host=$(TC_TARGET)
 
-ifeq ($(findstring $(ARCH),$(ARMv8_ARCHS)),$(ARCH))
-ENV += x264_ARCH=ARM
-CONFIGURE_ARGS += --host=aarch64-linux
+# Define x86asm or use gcc as assembler
+ifeq ($(findstring $(ARCH),$(i686_ARCHS) $(x64_ARCHS)),$(ARCH))
+NASM_BINARY = $(shell which nasm)
+ifeq ($(NASM_BINARY),)
+$(error nasm not found. Please install NASM assembler)
+endif
+ENV += AS=$(NASM_BINARY)
+else
```

Comment:
Similarly to ffmpeg, isn't this too much as nasm is always installed?
If anything, I'd rather have that test being done outside from individual cross but within the framework to avoid code duplication if feasible.